app/uk/gov/hmrc/helptosavestub/controllers/DESController.scala

```diff
@@ -16,25 +16,32 @@
 
 package uk.gov.hmrc.helptosavestub.controllers
 
-import cats.instances.list._
-import cats.instances.string._
-import cats.syntax.eq._
 import play.api.mvc.{Action, AnyContent, Request, Result}
+import uk.gov.hmrc.helptosavestub.config.AppConfig
 import uk.gov.hmrc.helptosavestub.util.Logging
 import uk.gov.hmrc.play.bootstrap.controller.BaseController
 
+import scala.collection.JavaConverters._
+
 // trait for controllers mimicking DES
-trait DESController { this: BaseController with Logging ⇒
+trait DESController {
+  this: BaseController with Logging with AppConfig ⇒
 
-  private val expectedAuthorisationHeader = List("Bearer test")
+  private val expectedDESHeaders: List[String] =
+    runModeConfiguration
+      .underlying
+      .getStringList("microservice.expectedDESHeaders")
+      .asScala
+      .toList
+      .map(e ⇒ s"Bearer $e")
 
   def desAuthorisedAction(body: Request[AnyContent] ⇒ Result): Action[AnyContent] = Action { request ⇒
-    val authorisationHeaders = request.headers.getAll("Authorization")
+    val authHeaders = request.headers.getAll("Authorization")
 
-    if (authorisationHeaders.toList === expectedAuthorisationHeader) {
+    if (expectedDESHeaders.containsSlice(authHeaders)) {
```

Comment:
why do we need `containsSlice` rather than `contains` here?

Reply:
because `authHeaders ` is a List

Reply:
taking this example:
```scala
val authHeaders                = List("a", "b", "c")
val expectedDESHeaders = List("a", "b")
```
we get
```scala
expectedDESHeaders.containsSlice(authHeaders) // = false
```
But this is probably ok for what we want since we're not expecting that sort of input in the request